odoo_module_migrate/migration.py

```diff
@@ -110,6 +110,12 @@ def _get_code_from_previous_branch(self, module_name, remote_name):
                 'version': target_version,
             }, path=self._directory_path)
 
+        logger.info("Getting latest changes from old branch")
+        # Depth is added just in case you had a shallow git history
+        _execute_shell(
+            "git fetch --depth 9999999 %s %s" % (remote_name, init_version)
```

Comment:
Can't you run with `--unshallow` instead?
```suggestion
            "git fetch --unshallow %s %s" % (remote_name, init_version)
```

Reply:
Using `--unshallow` would make the command fail if the git repo wasn't shallow.